pkg/connector/group.go

```diff
@@ -117,48 +117,109 @@ func (o *groupResourceType) Grants(
 	resource *v2.Resource,
 	token *pagination.Token,
 ) ([]*v2.Grant, string, annotations.Annotations, error) {
-	bag, page, err := parsePageToken(token.Token, &v2.ResourceId{ResourceType: resourceTypeGroup.Id})
+	var rv []*v2.Grant
+	bag := &pagination.Bag{}
+	err := bag.Unmarshal(token.Token)
 	if err != nil {
-		return nil, "", nil, fmt.Errorf("okta-connectorv2: failed to parse page token: %w", err)
+		return nil, "", nil, err
 	}
 
-	var rv []*v2.Grant
-	qp := queryParams(token.Size, page)
+	if bag.Current() == nil {
+		bag.Push(pagination.PageState{
+			ResourceTypeID: resourceTypeRole.Id,
+		})
+		bag.Push(pagination.PageState{
+			ResourceTypeID: resourceTypeUser.Id,
+		})
```

Comment:
super minor nit 😅 we are listing roles before users in the pagination bag but then the switch is the reverse